cli/internal/dataplane/bufferblob.go

```diff
@@ -59,15 +64,182 @@ type BufferEndMetadata struct {
 }
 
 type Container struct {
-	*url.URL
+	lock      *sync.Mutex
+	accessUrl *url.URL
+	getNewUrl func(context.Context) (*url.URL, error)
 }
 
-func clearBit(value int64, pos int) int64 {
-	mask := int64(^(1 << pos))
-	return value & mask
+func NewContainer(accessUrl *url.URL) *Container {
+	return &Container{accessUrl: accessUrl, lock: &sync.Mutex{}}
 }
 
-func (c *Container) GetBlobUrl(blobNumber int64) string {
+func NewContainerFromAccessString(ctx context.Context, accessString string) (*Container, error) {
+	accessUrl, err := ParseBufferAccessUrl(accessString)
+	if err != nil {
+		// Invalid URL, assume it's a file containing the URL
+		return NewContainerFromAccessFile(ctx, accessString)
+	}
+
+	// Valid URL, need to parse the SAS parameters in order to request a new URL later
+	bufferId := path.Base(accessUrl.Path)
+	qv := accessUrl.Query()
+	permissions := qv.Get("sp")
+	writeable := strings.Contains(permissions, "c")
+	st, err := parseSasQueryTimestamp(accessUrl, "st")
+	if err != nil {
+		return nil, err
+	}
+	se, err := parseSasQueryTimestamp(accessUrl, "se")
+	if err != nil {
+		return nil, err
+	}
+	accessTtl := ""
+	d := se.Sub(st)
+	if d > 0 {
+		accessTtl = common.DurationToTimeToLive(d).String()
+	}
+
+	getNewUrl := func(ctx context.Context) (*url.URL, error) {
```

Comment:
The purpose of being able to pass in a SAS URL directly is so that you don't have to be logged in to Tyger in order to read/write to a buffer. It would be worth checking to make sure the failure is graceful when we try to refresh an access URL in that case.

Reply:
We could just not refresh in this case